ide/coqide/wg_MessageView.ml

```diff
@@ -78,15 +90,82 @@ let message_view () : message_view =
     let mark = `MARK mark in
     let width = Ideutils.textview_width view in
     Ideutils.insert_xml ~mark buffer ~tags (Richpp.richpp_of_pp width msg);
-    buffer#insert ~iter:(buffer#get_iter_at_mark mark) "\n"
+    buffer#insert ~iter:(buffer#get_iter_at_mark mark) "\n";
+    buffer#move_mark (`NAME "end_of_output") ~where:buffer#end_iter;
+  in
+
+  (* Insert a prompt and make the buffer editable. *)
+  let insert_ltac_debug_prompt msg =
+    let tags = [] in
+    let mark = `MARK mark in
+    let width = Ideutils.textview_width view in
+    Ideutils.insert_xml ~mark buffer ~tags (Richpp.richpp_of_pp width msg);
+    buffer#move_mark (`NAME "end_of_output") ~where:buffer#end_iter;
+    view#set_editable true;
+    view#set_cursor_visible true;
+    view#scroll_to_mark (`NAME "end_of_output"); (* scroll to end *)
+    buffer#move_mark `INSERT ~where:buffer#end_iter;
+    let ins = buffer#get_iter_at_mark `INSERT in
+    buffer#select_range ins ins;  (* avoid highlighting *)
   in
 
+  let insert_msg = function
+    | Fb (lvl,msg) -> insert_fb_msg (lvl,msg)
+    | Prompt msg -> insert_ltac_debug_prompt msg
+
+  in
+  let msgs : message_entry_kind list ref = ref [] in
+  let (return, _) = GtkData.AccelGroup.parse "Return" in
+  let backspace = 65288 (*GtkData.AccelGroup.parse "Backspace"*) in
```

Comment:
How do we know this is stable across architectures? And more, why isn't this done just like in the two lines around?

Reply:
Got it.  I had to check the GTK source.  I didn't expect it to be "BackSpace" rather than "Backspace".